sklearn/utils/estimator_checks.py

```diff
@@ -1705,3 +1705,36 @@ def check_classifiers_regression_target(name, Estimator):
     e = Estimator()
     msg = 'Unknown label type: '
     assert_raises_regex(ValueError, msg, e.fit, X, y)
+
+
+@ignore_warnings(category=DeprecationWarning)
+def check_decision_proba_consistency(name, Estimator):
+    """
+    Check whether an estimator having both decision_function and
+    predict_proba methods has outputs with perfect rank correlation.
+    """
+
+    X, Y = make_multilabel_classification(n_classes=2, n_labels=1,
+                                          allow_unlabeled=True,
+                                          random_state=1)
+
+    estimator = Estimator()
+
+    set_testing_parameters(estimator)
+
+    if hasattr(estimator, "decision_function"):
+
+        if hasattr(estimator, "predict_proba"):
+            try:
+                classif = OneVsRestClassifier(estimator)
+                classif.fit(X, Y)
+                a = classif.predict_proba([i for i in range(20)])
+                b = classif.decision_function([i for i in range(20)])
+                assert_equal(
+                 rankdata(a, method='average'), rankdata(b, method='average'))
```

Comment:
`method` shouldn't matter as long as tied values have tied ranks. But if we're working with non-binary classification, we need to do this comparison column-wise. Use `assert_array_equal` rather than `assert_equal`.